test/assertions_test.rb

```diff
@@ -87,13 +87,43 @@ def test_assert_statsd_call
       end
     end
 
+    assert_no_assertion_triggered do
+      foo_1_metric_tags = { times: 2, tags: ['foo:1'] }
+      foo_2_metric_tags = { times: 1, tags: ['foo:2'] }
+      @test_case.assert_statsd_increments('counter', [foo_1_metric_tags, foo_2_metric_tags]) do
+        StatsD.increment('counter', tags: { foo: 1 })
+        StatsD.increment('counter', tags: { foo: 1 })
+        StatsD.increment('counter', tags: { foo: 2 })
+      end
+    end
+
     assert_assertion_triggered do
       @test_case.assert_statsd_increment('counter', times: 2, tags: ['foo:1']) do
         StatsD.increment('counter', tags: { foo: 1 })
         StatsD.increment('counter', tags: { foo: 2 })
       end
     end
 
+    assert_assertion_triggered do
+      foo_1_metric_tags = { times: 2, tags: ['foo:1'] }
+      foo_2_metric_tags = { times: 1, tags: ['foo:2'] }
+      @test_case.assert_statsd_increments('counter', [foo_1_metric_tags, foo_2_metric_tags]) do
+        StatsD.increment('counter', tags: { foo: 1 })
+        StatsD.increment('counter', tags: { foo: 2 })
+      end
+    end
+
+    assert_assertion_triggered do
+      foo_1_metric_tags = { times: 2, tags: ['foo:1'] }
+      foo_2_metric_tags = { times: 1, tags: ['foo:2'] }
+      @test_case.assert_statsd_increments('counter', [foo_1_metric_tags, foo_2_metric_tags]) do
+        StatsD.increment('counter', tags: { foo: 1 })
+        StatsD.increment('counter', tags: { foo: 1 })
+        StatsD.increment('counter', tags: { foo: 2 })
+        StatsD.increment('counter', tags: { foo: 2 })
+      end
+    end
```

Comment:
Instead of adding more and more assertions to this test case, I rather add a separate one (or multiple) with descriptions that tell me exactly the behavior that is being tested..